spotxBidAdapter2.js

```diff
@@ -0,0 +1,362 @@
+        let adId = null;
+        for (var i = 0; i < bidResponses.length; i++) {
+          if (bidResponses[i].requestId == spotxBid.bidId) {
+            adId = bidResponses[i].cache_key;
+          }
+        };
+        if (adId != null) {
+          spotxBid.bidId = adId;
+        }
+      });
+    }
+
+    return bidResponses;
+  },
+
+  /**
+   * If the publisher allows user-sync activity, the platform will call this function and the adapter may register pixels and/or iframe user syncs.
+   *
+   */
+  getUserSyncs: function(syncOptions, serverResponses) {},
+
+  /**
+   * If the adapter timed out for an auction, the platform will call this function and the adapter may register timeout.
+   *
+   */
+  onTimeout: function(timeoutData) {}
+}
+
+function outstreamRender(bid) {
+  if (bid.renderer.config.outstreamFunction != null && typeof bid.renderer.config.outstreamFunction === 'function') {
+    bid.renderer.config.outstreamFunction();
+    return;
+  }
+
+  window.console.log('[SPOTX][renderer] Handle SpotX outstream/inbanner renderer');
+
+  var script = window.document.createElement('script');
+  script.type = 'text/javascript';
+  script.src = '//search.spotxchange.com/js/' + bid.channel_id + '.js';
+  script.setAttribute('data-spotx_channel_id', '' + bid.channel_id);
+  script.setAttribute('data-spotx_vast_url', '' + bid.vastUrl);
+  script.setAttribute('data-spotx_content_width', bid.renderer.config.player_width);
+  script.setAttribute('data-spotx_content_height', bid.renderer.config.player_height);
+  script.setAttribute('data-spotx_content_page_url', bid.renderer.config.content_page_url);
+
+  window.parent.mySpotXAdDoneFunction = typeof bid.spotxAdDoneFunction === 'function' ? bid.spotxAdDoneFunction : function mySpotXAdDoneFunction(spotxAdFound) {
+    if (spotxAdFound) {
+      window.console.log('[SPOTX][renderer] ad playing here');
+    } else {
+      window.console.log('[SPOTX][renderer] code to place backup ad request here');
+    }
+  };
+  script.setAttribute('data-spotx_ad_done_function', 'mySpotXAdDoneFunction');
+
+  if (bid.renderer.config.ad_mute) {
+    script.setAttribute('data-spotx_ad_mute', '0');
+  }
+
+  if (bid.renderer.config.outstream_static) {
+    script.setAttribute('data-spotx_ad_unit', 'instream');
+    script.setAttribute('data-spotx_ad_skippable', '0');
+    script.setAttribute('data-spotx_custom_skin', '1');
+  } else {
+    script.setAttribute('data-spotx_ad_unit', 'incontent');
+    script.setAttribute('data-spotx_collapse', '0');
+  }
+
+  script.setAttribute('data-spotx_autoplay', '1');
+  // script.setAttribute('data-spotx_blocked_autoplay_override_mode', '1');
+  script.setAttribute('data-spotx_video_slot_can_autoplay', '1');
+
+  if (bid.renderer.config.inIframe && window.document.getElementById(bid.renderer.config.inIframe).nodeName == 'IFRAME') {
+    var rawframe = window.document.getElementById(bid.renderer.config.inIframe);
```

Comment:
Why are we using var here all of the sudden?

Reply:
https://medium.com/javascript-scene/javascript-es6-var-let-or-const-ba58b8dcde75